Marlin/Marlin_main.cpp

```diff
@@ -4220,6 +4229,54 @@ inline void gcode_M206() {
 
 #endif // FWRETRACT
 
+#ifdef ENABLE_M211
+/**
+ * M211: Enable or disable software endstops
+ */
+inline void gcode_M211() {
+  // manage bit mask in enable_software_endstops
+  uint8_t smask = 0x00;
+
+  if (code_seen_str("X0")) smask |= 1;
+  if (code_seen_str("Y0")) smask |= 2;
+  if (code_seen_str("Z0")) smask |= 4;
+  if (code_seen_str("X1")) smask |= 16;
+  if (code_seen_str("Y1")) smask |= 32;
+  if (code_seen_str("Z1")) smask |= 64;
+
+
+  // Set mask
+  if (code_seen('S'))
+  {
+     if (code_value_long()==0) {
+        // clear mask
+        enabled_software_endstops &= ~smask;
+     }
+     else {
+        // set mask
+        enabled_software_endstops |=  smask;
+     }
+
+  }
+  else {
+    // Print state
+  SERIAL_PROTOCOLLN(MSG_M211_REPORT);
+  SERIAL_PROTOCOLPGM(MSG_X_MIN);
+  SERIAL_PROTOCOLLN(((enabled_software_endstops & 1) > 0?MSG_ENABLED:MSG_DISABLED));
```

Comment:
We also have a handy `TEST` macro that can make these slightly clearer, e.g.:

    SERIAL_PROTOCOLLN(TEST(enabled_software_endstops, 0) ? MSG_ENABLED : MSG_DISABLED);